airbyte-integrations/connectors/source-slack/source_slack/source.py

```diff
@@ -155,9 +168,30 @@ def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[
         params.update(**stream_slice)
         return params
 
+    def parse_response(self, response: requests.Response, stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        for record in super().parse_response(response, **kwargs):
+            if "ts" in record:
+                record["ts"] = pendulum.from_timestamp(float(record["ts"]))
+
+            yield record
+
     def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]):
         current_stream_state = current_stream_state or {}
-        current_stream_state["ts"] = max(float(latest_record["ts"]), current_stream_state.get("ts", 0))
+        if isinstance(latest_record["ts"], DateTime):
+            latest_record["ts"] = latest_record["ts"].timestamp()
+
+        if isinstance(current_stream_state.get("ts", 0), str):
```

Comment:
What this block for?

Reply:
To cast timestamps and dates in needed format and back as we will have pendulum parsing error otherwise.

Reply:
I thought we removed float timestamps with `parse_response` above

Reply:
yeah, but we need return them back to time-stamps to keep consistency